### DISCUSSION TITLE
TLS 1.3 Handshake Type Extensions with OpenSsl

### DISCUSSION THREAD

**Discussion by dbeken85** [openssl/openssl#25413] [Q&A] [answered]
Hello,

I am using Freeradius and Openssl to perform TLS 1.3 handshake. 

I want to add the OID-Filter extension of TLS 1.3 into the Certificate Request Message during an TLS Handshake with mutual authentication. I cannot find a specific extension for OID_Filter in the source code.

How can I add extend the CertificateRequest message with this extension?

Is it a configuration I need to do in the openssl configuration? Or is it the application (freeradius) which needs to be extended? How is this done?

Regards,
Dincer

**Comment by mattcaswell** [comment, marked as answer]
OpenSSL does not support that extension.

It may be possible to achieve what you need using OpenSSL's custom extension's API which allows to you to add some basic support for non-builtin extensions, e.g. see this man page:

https://docs.openssl.org/3.4/man3/SSL_extension_supported/

But to make use of that API the freeradius application would need to be modified to support calling this API.

Otherwise you would have to add the support directly into OpenSSL itself.
